### PR TITLE
[#160195079] Add logic to display accurate data

### PR DESCRIPTION
**Pull request by Arthur236**
## What does this PR do?

displays correct data related to assets on the dashboard

## Description of Task to be completed?

```gherkin
Scenario: Accurate data on home page
Given an admin
When I visit the homepage
Then I should see correct data analysis (in summary) i.e lost, damaged, available, allocated.
```

## How should this be manually tested?

* Log in
* View changes in the data displayed about assets

## What are the relevant pivotal tracker stories?

[160195079](https://www.pivotaltracker.com/story/show/160195079)

## Any background context you want to add?

N/A

## Important notes

N/A

## Packages installed

N/A

## Deployment note

N/A

## Related PRs branch | PR (branch_name) | (pr_link)

N/A

## Todos
- [x] Raise PR
- [ ] Test
- [ ] Documentation

### CODE REVIEW

File: src/_reducers/assetStatus.reducer.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import constants from '../_constants';
+import initialState from './initialState';
+
+const {
+  ASSET_STATUS_LOADING,
+  ASSET_STATUS_FETCH_SUCCESS,
+  ASSET_STATUS_FETCH_FAILURE
+} = constants;
+
+export default(state = initialState.assetStatus, action) => {
```
Comment (sachungo) [inline src/_reducers/assetStatus.reducer.js]: Shouldn't there be a space between `default` and `(`?  That is,
```js
export default (state = initialState.assetStatus, action) => {
```
You can verify this in the [MDN docs](https://developer.mozilla.org/en-US/docs/web/javascript/reference/statements/export#Description)

File: src/components/AssetSpecs/AddAssetSpecComponent.jsx [hunk 2]
```diff
@@ -24,7 +24,7 @@ const AddAssetSpecComponent = (props) => {
   return (
     <Form onSubmit={props.handleSubmit}>
       <Form.Field>
-        <label htmlFor="year_of_manufacture" className="label-style">Year of Manufacture</label>
+        <div className="label-style">Year Of Manufacture</div>
```
Comment (sachungo) [inline src/components/AssetSpecs/AddAssetSpecComponent.jsx]: Why the change from `label` to `div`? `label` usually helps with accessibility. Please read more about [semantic HTML](https://developer.mozilla.org/en-US/docs/Learn/Accessibility/HTML) and their role in accessibility

File: src/components/Dashboard/AssetStatusComponent.jsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  }
+
+  render() {
+    const { asset, status } = this.props;
+    const activeClass = asset.count > 0 ? 'active-analytics' : '';
+
+    return (
+      <Grid.Column>
+        <div className={`analytics-state-rectangle ${activeClass}`}>
+          {asset && (
```
Comment (sachungo) [inline src/components/Dashboard/AssetStatusComponent.jsx]: Given that `asset` is an object and objects are always _truthy_, the checkmark will be shown even for status with a count of 0. I recommend that you assign `asset.count > 0` to a variable and use that. I.e.
```jsx
const hasAssets = asset.count > 0;
const activeClass = hasAssets ? 'active-analytics': '';

return (
  ....
  {hasAssets && (

File: src/components/Dashboard/DashboardComponent.jsx [hunk 4]
```diff
@@ -45,6 +43,10 @@ class DashboardComponent extends Component {
   ];
 
   render() {
+    const {
+      allocatedAssets, availableAssets, damagedAssets, lostAssets
```
Comment (sachungo) [inline src/components/Dashboard/DashboardComponent.jsx]: If you want to use multiline, then place each "destructured" value in its own line. I.e.
```js
const {
  allocatedAssets,
  availableAssets,
  damagedAssets,
  lostAssets
} = this.props;
```

File: src/components/Dashboard/DashboardComponent.jsx [hunk 5]
```diff
@@ -149,4 +99,38 @@ class DashboardComponent extends Component {
...
+};
+
+DashboardComponent.defaultProps = {
+  lostAssets: {},
+  availableAssets: {},
+  damagedAssets: {},
+  allocatedAssets: {}
+};
+
+const mapStateToProps = ({ assetStatus }) => {
```
Comment (sachungo) [inline src/components/Dashboard/DashboardComponent.jsx]: Why haven't you separated `mapStateToProps` and the `connect` part to a container file? Remember that the component being connected to doesn't have to be in the same file as `mapStateToProps` and `mapDispatchToProps`. I'm suggesting the separation as this section should be part of the containers folder as explained [here](https://github.com/AndelaOSP/art-dashboard#containers-folder-_components) and it mostly entails copy pasting the logic.

You can also check [this redux example](https://redux.js.org/basics/exampletodolist#containers-filterlink-js)

File: src/components/AssetSpecs/AddAssetSpecComponent.jsx (hunk 2 above)
Comment (Arthur236) [reply src/components/AssetSpecs/AddAssetSpecComponent.jsx]: This is how the previous forms have been implemented. I think we should have a chore to change the rest as well.

File: src/_components/Dashboard/DashboardContainer.jsx [hunk 6]
```diff
@@ -0,0 +1,63 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+import { getAssetStatus } from '../../_actions/assetStatus.action';
+
+import DashboardComponent from '../../components/Dashboard/DashboardComponent';
+
+class DashboardContainer extends PureComponent {
```
Comment (sachungo) [inline src/_components/Dashboard/DashboardContainer.jsx]: This is a redundant component as it is just passing props to `DashboardComponent`. Please delete it

File: src/_components/Dashboard/DashboardContainer.jsx [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    allocatedAssets: allocated,
+    availableAssets: available,
+    damagedAssets: damaged,
+    lostAssets: lost
+  };
+};
+
+export default withRouter(connect(mapStateToProps, {
+  getAssetStatus
+})(DashboardContainer));
```
Comment (sachungo) [inline src/_components/Dashboard/DashboardContainer.jsx]: Use `DashboardComponent` here

**Comment by sachungo** [comment]
@Arthur236 once you feel better, resolve the conflicts and merge the PR
